src/main/java/org/generationcp/middleware/dao/GermplasmStudySourceDAO.java

```diff
@@ -52,7 +56,16 @@
 		+ "LEFT JOIN ims_lot lot ON lot.eid = gss.gid \n"
 		+ "WHERE gss.project_id = :studyId ";
 
-	public List<GermplasmStudySourceDto> getGermplasmStudySourceList(final GermplasmStudySourceSearchRequest germplasmStudySourceSearchRequest) {
+	public Map<Integer, GermplasmStudySource> getGermplasmStudySourcesMap(final Set<Integer> gids) {
+		final Criteria criteria = this.getSession().createCriteria(this.getPersistentClass());
+		criteria.createAlias("germplasm", "germplasm");
```

Comment:
Do we have a unique key on `germplasm_study_source.gid` field?

Reply:
I created a new changeset to add unique index constraint for germplasm_study_source.gid.